hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedRocksIterator.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.hadoop.hdds.utils.db.managed;
+
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.RocksIterator;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Managed RocksIterator.
+ */
+public class ManagedRocksIterator extends RocksIterator {
+  private final RocksIterator original;
+
+  public ManagedRocksIterator(RocksDB parent, RocksIterator original) {
+    // This object just act as a proxy to the real iterator.
+    super(parent, original.getNativeHandle());
```

Comment:
There are two RocksIterator objects which may cause problems in the future.  E.g. RocksDB may add a new method to RocksIterator but it is not overridden in ManagedRocksIterator.  I suggest to have a get() method as below.
```
public class ManagedRocksIterator {
  private final RocksIterator original;

  public ManagedRocksIterator(RocksIterator original) {
    this.original = original;
  }

  public RocksIterator get() {
    return original;
  }

  @Override
  protected void finalize() throws Throwable {
    ManagedRocksObjectUtils.assertClosed(original);
    super.finalize();
  }
}
```

Reply:
Good one, this make the code a lot simpler actually.